### PR TITLE
Streamer update for the new DXFeed API

### PR DESCRIPTION
**Pull request by Quenos**
## Description
DXFeed introduced a new API 
- streamer.py and dxFeed classes updated to work with new API

All contracts stayed the same so code should be working, albeit that at current this new API has only limited capabilities.
No options e.g. Also some symbols are not working

## Related issue(s)
No issue submitted

## Pre-merge checklist
- [x] Passing tests:
        - Happy flow and some faults are tested
           * candles are not working, also not working with the example of the DXfeed website.
- [ ] New tests added (if applicable)

### CODE REVIEW

**Comment by Quenos** [comment]
It's a bit beyond me where to put
```
import requests
```
in order to pass the lint for the sequence of the imports

File: tastytrade/metrics.py
Comment (Quenos) [inline tastytrade/metrics.py:1]: This file was not touched and should not be integrated.
Unsure what happened

**Comment by Graeme22** [comment]
Hey! This looks interesting, I'm excited to try it out! I noticed the other streamer in the API docs and I am intrigued.

To start off, could you make this in addition to the existing streamer instead of replacing it? I don't think it makes sense to replace the existing streamer yet, especially when the new one has more limited functionality.

Other than that, I think it would be good to get the tests passing. And the `Event` classes probably don't need to be modified; the event type is implicit in the class definition. Maybe remove that key from the dict before casting it into a class object?

> It's a bit beyond me where to put
> 
> ```
> import requests
> ```
> 
> in order to pass the lint for the sequence of the imports

You can just run `isort tastytrade` and the order will be fixed automagically.

**Comment by Quenos** [comment]
- old streamer put back. Switchable using use_legacy flag on create
- Used isort, but checker still complaining. Is there a way to check this locally before pushing?

**Comment by Quenos** [comment]
Need some help with these last three issues

File: setup.py [hunk 1]
```diff
@@ -1,6 +1,5 @@
 from setuptools import find_packages, setup
 
-
```
Comment (Graeme22) [inline setup.py:3]: Can we get rid of all the files that weren't actually changed?

File: tastytrade/dxfeed/candle.py [hunk 2]
```diff
@@ -10,6 +10,7 @@ class Candle(Event):
     for a specific period. Candles are build with a specified period using a
     specified price type with data taken from a specified exchange.
     """
+
     #: symbol of this event
```
Comment (Graeme22) [inline tastytrade/dxfeed/candle.py:14]: Like this one

File: tastytrade/streamer.py [hunk 3]
```diff
@@ -561,6 +586,387 @@ async def _map_message(self, message) -> None:
...
+
+        #: The unique client identifier received from the server
+        self._session = session
+        self._authenticated = False
+        api_token = self.get_api_token()
+        self._auth_token = api_token['token']
+        self._wss_url = api_token['dxlink-url']
+
+        self._connect_task_new = asyncio.create_task(self._connect())
+        self._keepalive = asyncio.create_task(self._keepalive())
```
Comment (Graeme22) [inline tastytrade/streamer.py:700]: The errors are because you're using `_keepalive` as the name of the method and the variable.

**Review by Graeme22 — CHANGES_REQUESTED** [review]
Looks good! Can you clean up a few more things?

**Comment by michaelcooke** [comment]
@Graeme22 @Quenos is there anything I can do to help push this one along? It looks like all of the remaining items were addressed.

**Comment by Graeme22** [comment]
Hey! I've been crazy busy the last couple months, but things are calmer now so I should get the chance to tackle this in the next couple days. Thanks for being patient in the meantime!

**Comment by Graeme22** [comment]
This is more urgent than I thought, it seems like the old streamer has been moved to delayed quotes! In light of this I don't think there's any reason to keep the old streamer.

**Comment by michaelcooke** [comment]
> it seems like the old streamer has been moved to delayed quotes!

It's still possible to get live quotes using dxFeed with the token and url obtained by the `/quote-streamer-tokens` endpoint. On the platform, you can switch from dxFeed to dxLink by opening settings and triple-clicking the "Currently using..." text in the general tab. Tastytrade hasn't told me how long they -- or dxFeed -- will support this data feed, but enough users are reporting issues with the websocket stream that I anticipate it will be supported for the platform for the time being.

I would bet that your quotes are being delayed arbitrarily by tastytrade; I had the same issue. You should stop any applications you personally have running using dxFeed, contact tastytrade support to complain about delayed quotes and mention you are not using the old data system, and wait for them to switch you back to live data. From there, you should be able to use dxFeed or dxLink for live data.

If it sounds confusing, arbitrary, and opaque, it's because it is.

**Comment by michaelcooke** [comment]
As a testing note, `wss://tasty-openapi-ws.dxfeed.com/realtime` supports fewer event types than `wss://tasty-live-ws.dxfeed.com/realtime`.

Supported events by both feeds:
- Quote
- Candle
- Trade
- Summary
- Profile

Undocumented events supported only by live feed for platform:
- DailyCandle
- TradeETH
- Greeks
- TimeAndSale
- Underlying
- AnalyticOrder
- OptionSale
- Order
- Series
- Configuration
- Message

**Comment by Graeme22** [comment]
> As a testing note, `wss://tasty-openapi-ws.dxfeed.com/realtime` supports fewer event types than `wss://tasty-live-ws.dxfeed.com/realtime`. Tastytrade has told me this is intentional, and were surprised to hear that the extra events listed below were supported by the live platform feed.
> 
> Supported events by both feeds:
> 
> * Quote
> * Candle
> * Trade
> * Summary
> * Profile
> 
> Undocumented events supported only by live feed for platform:
> 
> * DailyCandle
> * TradeETH
> * Greeks
> * TimeAndSale
> * Underlying
> * AnalyticOrder
> * OptionSale
> * Order
> * Series
> * Configuration
> * Message

This is great stuff, thanks for researching this!

**Comment by Graeme22** [comment]
> It's still possible to get live quotes using dxFeed with the token and url obtained by the `/quote-streamer-tokens` endpoint. On the platform, you can switch from dxFeed to dxLink by opening settings and triple-clicking the "Currently using..." text in the general tab. Tastytrade hasn't told me how long they -- or dxFeed -- will support this data feed, but enough users are reporting issues with the websocket stream that I anticipate it will be supported for the platform for the time being.

That's good to know, so we have more time to get these changes integrated then. That said, we should probably get this PR moving along.

**Comment by Graeme22** [comment]
To start, can we name the streamers `DXFeedStreamer` and `DXLinkStreamer`? And update the DXLink streamer to only support the events that are currently working?

**Comment by Quenos** [comment]
What is still needed here, lost track due to all the comments.

**Comment by Graeme22** [comment]
> To start, can we name the streamers `DXFeedStreamer` and `DXLinkStreamer`? And update the DXLink streamer to only support the events that are currently working?

This to start with

**Pull request merged**